### DISCUSSION TITLE
Simplify print number procedure

### DISCUSSION THREAD

**Discussion by Alexey1994** [klange/toaruos#262] [Ideas]
https://github.com/klange/toaruos/blob/a24baa2b58e4a52ecb71174843a6718bc86ff70c/boot/text.c#L262

`void print_uint_(unsigned int n)
{
  if(n/10) {
    print_uint_(n/10);
  }

  putchar(n%10 + '0');
}`

**Comment by klange** [comment]
The main issue with your proposed implementation is that it is recursive and will use more space in stack frames than the existing implementation uses in creating a fixed-size buffer.

It is also worth noting that the existing `print_int_` function in the bootloader is derived from [one found in the libc](https://github.com/klange/toaruos/blob/master/libc/stdio/printf.c#L6) and has a complex history; much of its seemingly odd design stems from support for padding and alignment options which have been removed in the bootloader version.

**Comment by Alexey1994** [comment]
Recursive in assembler makes one div per character, and div is a very slow operation

**Comment by klange** [comment]
This whole discussion is moot as this function is never used. The one place where the bootloader prints arbitrary decimal integers is the video mode selection screen, which uses a different implementation.

**Reply by Alexey1994** [reply]
I'm just sharing my experience, the choice is yours (:
